tests/subset.rs

```diff
@@ -42,24 +42,43 @@ fn test_subset<A: Adversary<Subset<NodeId>>>(
     while !network.nodes.values().all(TestNode::terminated) {
         network.step();
     }
+
+    let subsetoutput_to_option = |x: &SubsetOutput<_>| {
+        match x.clone() {
+            SubsetOutput::Contribution(x, y) => Some((x, y)),
+            SubsetOutput::Done               => None
+        }
+    };
+
     // Verify that all instances output the same set.
-    let mut expected = None;
+    let mut observer: Vec<_> = network.observer.outputs().iter().map(subsetoutput_to_option).collect();
+    observer.sort();
```

Comment:
Instead of sorting, you could collect it as a `BTreeSet` or `HashSet` (or a map).

(Good idea, though, to replace `expected` with the observer output! Much simpler. :+1: )

Reply:
Thank you!

The most annoying part of this is that `SubsetOutput` doesn’t implement `Ord`, and if I add such a requirement for `T` then it bubbles up into the public API.  So I am forced to use `subsetoutput_to_option`.

Reply:
Since you mentioned it, you could simplify the test by deriving `PartialOrd` and `Ord` for `SubsetOutput`. The derived instances use the lexicographic ordering, so `Done` is greater than `Contribution`.

Reply:
Also see the Rust convention [C-COMMON-TRAITS](https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits). We adopted the standard conventions for `hbbft`.

Reply:
@vkomenda I tried deriving `Ord` but it resulted in compiler errors.  I didn’t think to derive `PartialOrd` as well until I looked at the Rust source code.

Reply:
Sorry, @vkomenda is right (see below), of course, and the `Vec` was probably a good idea.
But since we also count the number of outputs, double entries should automatically cause the test to fail, too, I guess, so it should be fine either way?

Reply:
That's right, there is a uniqueness check in the test when a new contribution is inserted into a map. I think the difference is only in testing the observer output which is converted to a set without this check. It's a minor detail and I'll leave it to @DemiMarie whether to take care of it or not.